packages/grafana-ui/src/components/PanelChrome/PanelContext.ts

```diff
@@ -82,6 +84,17 @@ export interface PanelContext {
    * Called when a panel is changing the sort order of the legends.
    */
   onToggleLegendSort?: (sortBy: string) => void;
+
+  /**
+   * Some panels could be interested in updating the queries.
+   * This optional and not always available in all contexts.
+   */
+  onUpdateQueries?: (queries: DataQuery[]) => void;
+
+  /**
+   * Optional, only some contexts support this. This can fail / be cancelled. Which is why it returns a promise.
+   */
+  onUpdateData?: (frames: DataFrame[]) => Promise<boolean>;
```

Comment:
I feel like the update that removes your queries needs a ConfirmModal as it's a destructive action where the user looses all his queries. Question who is responsible for that confirm modal? the panel or the context implementation?

Reply:
I think the context should own confirming the destructive action, since leaving it to panels means that panel devs might not always enforce it.